sklearn/linear_model/_quantile.py

```diff
@@ -127,7 +129,11 @@ def fit(self, X, y, sample_weight=None):
             Returns self.
         """
         X, y = self._validate_data(
-            X, y, accept_sparse=False, y_numeric=True, multi_output=False
+            X,
```

Comment:
we need to update the documentation of `fit` and the different `predict` such that we mention `{array-like, sparse matrix}`. In addition, in the description, we should mention that CSC and CSR are accepted without additional copies while other sparse formats will be converted to CSR.